File: java/src/mpc/MPCDesignationCLI.java
package mpc;

import java.util.ArrayList;
import java.util.List;

/**
 * Command-line interface for MPC designation converter.
 */
public class MPCDesignationCLI {

    private static void printUsage() {
        System.err.println("Usage: mpc_designation [-v|--verbose] <designation> [designation ...]");
        System.err.println();
        System.err.println("Convert between packed and unpacked MPC designations.");
        System.err.println("Auto-detects the input format and converts to the other.");
        System.err.println();
        System.err.println("Options:");
        System.err.println("  -v, --verbose   Show detailed information about the conversion");
        System.err.println("  --version       Show version information");
        System.err.println();
        System.err.println("Examples:");
        System.err.println("  mpc_designation 00001             -> 1");
        System.err.println("  mpc_designation 1                 -> 00001");
        System.err.println("  mpc_designation J95X00A           -> 1995 XA");
        System.err.println("  mpc_designation '1995 XA'         -> J95X00A");
        System.err.println("  mpc_designation 'C/1995 O1'       -> CJ95O010");
        System.err.println("  mpc_designation 1P                -> 0001P");
    }

    public static void main(String[] args) {
        if (args.length == 0) {
            printUsage();
            System.exit(1);
        }

        boolean verbose = false;
        List<String> designations = new ArrayList<>();

        for (String arg : args) {
            switch (arg) {
                case "-v":
                case "--verbose":
                    verbose = true;
                    break;
                case "-h":
                case "--help":
                    printUsage();
                    System.exit(0);
                    break;
                case "--version":
                    System.out.println("mpc_designation " + MPCDesignation.VERSION);
                    System.exit(0);
                    break;
                default:
                    designations.add(arg);
                    break;
            }
        }

        if (designations.isEmpty()) {
            printUsage();
            System.exit(1);
        }

        boolean multiple = designations.size() > 1;

        for (String des : designations) {
            try {
                MPCDesignation.Result result = MPCDesignation.convert(des);
                MPCDesignation.Info info = result.info;

                if (verbose) {
                    System.out.println("  Input:    " + des);
                    System.out.println("  Detected: " + info.format + " format, " + info.subtype);
                    String action = info.format == MPCDesignation.FormatType.PACKED
                        ? "unpacking to human-readable form"
                        : "packing to MPC compact form";
                    System.out.println("  Action:   " + action);
                    System.out.println("  Output:   " + result.output);
                    if (multiple) {
                        System.out.println();
                    }
                } else if (multiple) {
                    System.out.println(des + " -> " + result.output);
                } else {
                    System.out.println(result.output);
                }
            } catch (MPCDesignation.MPCDesignationException e) {
                System.err.println("Error: " + e.getMessage());
                System.exit(1);
            }
        }
    }
}

File: octave/src/mpc_designation_cli.m
#!/usr/bin/env octave
% MPC Designation CLI - Command-line interface
%
% Usage: octave --no-gui mpc_designation_cli.m [options] designation [designation ...]
%
% Options:
%   -v, --verbose   Show detailed information
%   -h, --help      Show help message

% Prevent running as function
1;

function print_usage()
  fprintf('Usage: octave --no-gui mpc_designation_cli.m [-v|--verbose] <designation> [designation ...]\n');
  fprintf('\n');
  fprintf('Convert between packed and unpacked MPC designations.\n');
  fprintf('Auto-detects the input format and converts to the other.\n');
  fprintf('\n');
  fprintf('Options:\n');
  fprintf('  -v, --verbose   Show detailed information about the conversion\n');
  fprintf('  -h, --help      Show this help message\n');
  fprintf('\n');
  fprintf('Examples:\n');
  fprintf('  octave --no-gui mpc_designation_cli.m 00001             -> 1\n');
  fprintf('  octave --no-gui mpc_designation_cli.m 1                 -> 00001\n');
  fprintf('  octave --no-gui mpc_designation_cli.m J95X00A           -> 1995 XA\n');
  fprintf('  octave --no-gui mpc_designation_cli.m ''1995 XA''         -> J95X00A\n');
end

% Source the library
script_dir = fileparts(mfilename('fullpath'));
if isempty(script_dir)
  script_dir = '.';
end
source(fullfile(script_dir, 'mpc_designation.m'));

% Get command line arguments
args = argv();

if length(args) == 0
  print_usage();
  exit(1);
end

verbose = false;
designations = {};

for i = 1:length(args)
  arg = args{i};
  if strcmp(arg, '-v') || strcmp(arg, '--verbose')
    verbose = true;
  elseif strcmp(arg, '-h') || strcmp(arg, '--help')
    print_usage();
    exit(0);
  else
    designations{end+1} = arg;
  end
end

if length(designations) == 0
  print_usage();
  exit(1);
end

multiple = length(designations) > 1;

for i = 1:length(designations)
  des = designations{i};
  try
    result = mpc_convert(des);
    info = result.info;
    output = result.output;

    if verbose
      fprintf('  Input:    %s\n', des);
      fprintf('  Detected: %s format, %s\n', info.format, info.subtype);
      if strcmp(info.format, 'packed')
        action = 'unpacking to human-readable form';
      else
        action = 'packing to MPC compact form';
      end
      fprintf('  Action:   %s\n', action);
      fprintf('  Output:   %s\n', output);
      if multiple
        fprintf('\n');
      end
    elseif multiple
      fprintf('%s -> %s\n', des, output);
    else
      fprintf('%s\n', output);
    end
  catch err
    fprintf(2, 'Error: %s\n', err.message);
    exit(1);
  end
end
